### ISSUE TITLE
GPL compatibility?

### ISSUE THREAD

**Issue by abraxa** [signalspec/signalspec#1]
Hi, sigrok dev here. Your project is very interesting and I can definitely see a use case for wrapping your library in, say, a generic protocol decoder. However, I'm not sure how integration could take place license-wise.

https://www.mozilla.org/en-US/MPL/2.0/FAQ/ (Question #14)
https://www.mozilla.org/en-US/MPL/2.0/combining-mpl-and-gpl/

What are your thoughts on this?

**Comment by kevinmehall** [comment]
Yeah, I'd love to see some kind of integration with Sigrok / Pulseview. They're currently what I use when I need something that actually works :wink:

My reading of that page (and my prior understanding) is that the MPL2 contains language specifically to ensure compatibility with GPL. My goal would be to design public APIs such that Signalspec could be connected to something like Sigrok easily and you'd be in the "Unmodified MPL-licensed Files" section of that page that doesn't require doing anything special.

The license choice is also not set in stone. Currently I'm the sole copyright holder, and I don't really expect anyone else to contribute until I can get it into a state where it does something useful. It concerns me a little that MPL2 is not one of the most popular licenses, but it seems like a good middle ground between GPL and Apache. I like that it forbids proprietary forks while putting a very clear boundary at the file level. LGPL's language regarding linking seems much murkier, especially when using Rust instead of C. I'd consider a more permissive license, though, especially for the "standard library" of protocol definitions.
